### PR TITLE
doctrine mappings: annotations --> attrributes

### LINKED ISSUE #209: replace annotation with attribute
Because Doctrine 3 removed support for
BC Break: Removed AnnotationDriver

replace with attributes

### PR DESCRIPTION
**Pull request by pinclau**
changed all entities that uses doctrine annotation mappings with doctrine attribute mappings
#209

### CODE REVIEW

File: src/User/src/Entity/User.php [hunk 1]
```diff
@@ -33,38 +32,34 @@ class User extends AbstractEntity implements UserEntityInterface
...
-    /** @ORM\Column(name="identity", type="string", length=191, unique=true) */
+    #[ORM\Column(name: "identity", type: "string", length: 191, unique: true)]
     protected string $identity;
 
-    /** @ORM\Column(name="password", type="string", length=191) */
+    #[ORM\Column(name: "password", type: "string", length: 191)]
     protected string $password;
 
-    /** @ORM\Column(name="status", type="string", length=20) */
+    #[ORM\Column("status", type: "string", length: 20)]
```
Comment (alexmerlin) [inline src/User/src/Entity/User.php]:
```suggestion
    #[ORM\Column(name: "status", type: "string", length: 20)]
```

File: src/User/src/Entity/User.php [hunk 2]
```diff
@@ -33,38 +32,34 @@ class User extends AbstractEntity implements UserEntityInterface
...
     protected string $status = self::STATUS_PENDING;
 
-    /** @ORM\Column(name="isDeleted", type="boolean") */
+    #[ORM\Column("isDeleted", type: "boolean")]
```
Comment (alexmerlin) [inline src/User/src/Entity/User.php]:
```suggestion
    #[ORM\Column(name: "isDeleted", type: "boolean")]
```

File: src/User/src/Entity/UserResetPasswordEntity.php [hunk 3]
```diff
@@ -11,11 +11,7 @@
 use Doctrine\ORM\Mapping as ORM;
 use Throwable;
 
-/**
- * @ORM\Entity()
- * @ORM\Table(name="user_reset_password")
- * @ORM\HasLifecycleCallbacks()
- */
+#[ORM\Entity, ORM\Table(name: "user_reset_password"), ORM\HasLifecycleCallbacks]
```
Comment (alexmerlin) [inline src/User/src/Entity/UserResetPasswordEntity.php]:
```suggestion
#[ORM\Entity]
#[ORM\Table(name: "user_reset_password")]
#[ORM\HasLifecycleCallbacks]
```
Just for better readability.

File: src/User/src/Entity/UserResetPasswordEntity.php [hunk 4]
```diff
@@ -25,19 +21,17 @@ class UserResetPasswordEntity extends AbstractEntity
         self::STATUS_REQUESTED,
     ];
 
-    /**
-     * @ORM\ManyToOne(targetEntity="User", cascade={"persist", "remove"}, inversedBy="resetPasswords")
-     * @ORM\JoinColumn(name="userUuid", referencedColumnName="uuid")
-     */
+    #[ORM\ManyToOne(targetEntity: "User", cascade: ['persist', 'remove'], inversedBy: "resetPasswords")]
```
Comment (alexmerlin) [inline src/User/src/Entity/UserResetPasswordEntity.php]:
```suggestion
    #[ORM\ManyToOne(targetEntity: User::class, cascade: ['persist', 'remove'], inversedBy: "resetPasswords")]
```

File: src/User/src/Entity/UserRole.php [hunk 5]
```diff
@@ -6,23 +6,24 @@
...
+#[ORM\Entity(repositoryClass: UserRoleRepository::class)]
+#[ORM\Table(name: "user_role")]
+#[ORM\HasLifecycleCallbacks]
 class UserRole extends AbstractEntity implements RoleInterface
 {
-    public const ROLE_GUEST = 'guest';
-    public const ROLE_USER  = 'user';
-    public const ROLES      = [
+    public const ROLE_GUEST    = 'guest';
+    public const ROLE_API_USER = 'api_user';
```
Comment (alexmerlin) [inline src/User/src/Entity/UserRole.php]:
```suggestion
```

File: src/User/src/Entity/UserRole.php [hunk 6]
```diff
@@ -6,23 +6,24 @@
...
+    public const ROLE_USER     = 'user';
+    public const ROLES         = [
         self::ROLE_GUEST,
+        self::ROLE_API_USER,
```
Comment (alexmerlin) [inline src/User/src/Entity/UserRole.php]:
```suggestion
```

**Review by alexmerlin — CHANGES_REQUESTED** [review]

**Review by alexmerlin — APPROVED** [review]

**Review by marioradu05 — APPROVED** [review]

**Pull request merged**